src/stream/src/common/table/state_table.rs

```diff
@@ -450,14 +445,10 @@ impl<S: StateStore> StateTable<S> {
             },
             None => {
                 assert!(pk.len() <= self.pk_indices.len());
-                let key_indices = (0..pk.len())
-                    .into_iter()
-                    .map(|index| self.pk_indices[index])
-                    .collect_vec();
                 let read_options = ReadOptions {
-                    dist_key_hint: None,
-                    check_bloom_filter: !self.dist_key_indices.is_empty()
-                        && self.dist_key_indices == key_indices,
+                    prefix_hint: None,
+                    check_bloom_filter: self.prefix_hint_len != 0
+                        && self.prefix_hint_len <= pk.len(),
```

Comment:
Same here. See the comment above.

Reply:
Also, for state table, we can add a sanity check to ensure bloom filter is used.

Reply:
sanity check is a bit stronger, let's send warning when `check_bloom_filter==false`

Reply:
sanity check is a bit stronger, let's send warning when `check_bloom_filter==false`